apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php

```diff
@@ -83,7 +83,22 @@ public function enqueue_script_and_style() {
 	private function should_not_show_callout() {
 		$blog_id = defined( 'IS_WPCOM' ) && IS_WPCOM ? get_current_blog_id() : \Jetpack_Options::get_option( 'id' );
 
-		return $this->blog_has_custom_domain( $blog_id ) || $this->user_has_email_unverified() || $this->blog_is_p2( $blog_id );
+		return $this->has_unlaunched_launchpad() || $this->blog_has_custom_domain( $blog_id ) || $this->user_has_email_unverified() || $this->blog_is_p2( $blog_id );
+	}
+
+	/**
+	 * Check for site-intent and not launched site on launchpad.
+	 */
+	private function has_unlaunched_launchpad() {
+		$site_intent      = get_option( 'site_intent' );
+		$task_statuses    = get_option( 'launchpad_checklist_tasks_statuses', array() );
+		$site_is_launched = ! isset( $task_statuses['site_launched'] ) || ! $task_statuses['site_launched'];
```

Comment:
You're right, but we have to change the if because we want to show the upsell to `unlaunched` sites but not while within the `launchpad`, like this:

```
$site_is_unlaunched = get_option( 'launch-status' ) === 'unlaunched';
if ( ! $site_intent || ( $site_intent && ! $site_is_unlaunched ) ) {
```

Reply:
Yeah, might could even do `$site_is_launched = get_option( 'launch-status' ) === 'launched';` to get rid of the `!` and make it more clear. I was thinking 'launch-status' might be a more definitive "source of truth" on whether the site is launched than 'launchpad_checklist_tasks_statuses['site_launched']'. But maybe there's a good reason to use `launchpad_checklist_tasks_statuses` instead?

Reply:
I was thinking, what happens if the user finished the process and launched the site in Launchpad but then passed it to `coming soon`?

Reply:
Oh, I see! I'm not sure if that sets 'launch-status' to 'unlaunched' or not. 🤔 But, that makes sense why you might want to use `launchpad_checklist_tasks_statuses` instead. Cool with me! ☺️ 🚀